[Security] 9.5.0 release notes (#7462)
elastic/docs-content: release-notes/elastic-security/index.md

* Adds `security.enableRule` and `security.disableRule` workflow steps for bulk enabling or disabling detection rules by rule IDs or query [#275187]({{kib-pull}}275187).
* Enables the **Close all alerts that match this exception** checkbox when creating an exception from a runtime field defined on the rule's source index [#270219]({{kib-pull}}270219).
* Adds a warning callout and confirmation modal to the rule exceptions form when entering escaped characters (such as `\*` or `\?`) with the `matches` operator, indicating they might have intended wildcards instead [#268397]({{kib-pull}}268397).
* Enables the advanced setting that turns on the new flyout system in {{elastic-sec}} [#279300]({{kib-pull}}279300).
* Enables search in the **JSON** tab of the Security Solution alert flyout, matching the behavior of the Discover document flyout [#263875]({{kib-pull}}263875).
* Adds an action in the correlations section of the alert details flyout to open a related attack as a preview (legacy expandable flyout) or as a child panel (new flyout system) [#279306]({{kib-pull}}279306).

> * Adds an **Add to chat** bulk action to the alerts table, letting analysts send up to 20 selected alerts to the {{agent-builder}} chat for triage [#270904]({{kib-pull}}270904).

Reviewer: This should be 100 alerts instead of 20. Confirmed on latest BC

Author: ```suggestion
* Adds an **Add to chat** bulk action to the alerts table, letting analysts send up to 100 selected alerts to the {{agent-builder}} chat for triage [#270904]({{kib-pull}}270904).
```

* Adds an Anomalies section to the entity details flyout and details tab for host and user entities, showing a 30-day anomaly overview by MITRE ATT&CK tactic and detailed, filterable anomaly records, with actions including **Add to Timeline**, **View in Discover**, and opening the **Single Metric Viewer** in {{ml}} [#273139]({{kib-pull}}273139), [#275118]({{kib-pull}}275118).
* Adds an `entityStore.updateAssetCriticality` workflow step that updates an entity's asset criticality and triggers a risk score recalculation [#276092]({{kib-pull}}276092).
* Adds `entityStore.entityAssetCriticalityUpdated` and `entityStore.entityRiskScoreChanged` workflow triggers that fire when an entity's asset criticality is set or cleared, or when its risk score changes [#275670]({{kib-pull}}275670).
* Updates Entity Store APIs to reject unknown keys in request bodies, query parameters, and path parameters with a `400 Bad Request` instead of silently ignoring them [#273098]({{kib-pull}}273098).
* Adds a rich visual renderer for `security.entity` attachments in {{agent-builder}}, displaying resolved entities as cards or tables [#264985]({{kib-pull}}264985).

> % TODO: Confirm with the PR author whether this belongs in the 9.5 release notes. This is gated behind the `entityAttachmentsEnabled` experimental feature flag, which defaults to `false` (OFF) in 9.5, so it may not be generally available to customers.

Author: Hi @opauloh, could you confirm if the entry on line 76 should be listed in the 9.5 release notes? Thanks!

Reviewer 2: Hi @natasha-moore-elastic , thanks for confirming, this was actually added on 9.4.0 (late merged just few days before the released upon leadership approval). So it seems to me that it should have been on 9.4.0 notes and not 9.5?

Author: Moved to 9.4.0 in [09157d8](https://github.com/elastic/docs-content/pull/7462/commits/09157d86b2878cd87a563be72957bb1cb3e79cf6).

* Adds recurrence-based scheduling to Osquery packs. Pack authors can run a pack on a **Daily** or **Custom** schedule, with an optional start date, stop-after date, and splay, and can override the schedule per query [#275896]({{kib-pull}}275896), [#271865]({{kib-pull}}271865), [#270639]({{kib-pull}}270639), [#269730]({{kib-pull}}269730).
* Adds the ability to export Osquery query results as CSV, JSON, or NDJSON files directly from the query results page [#275597]({{kib-pull}}275597), [#267800]({{kib-pull}}267800), [#266582]({{kib-pull}}266582), [#265995]({{kib-pull}}265995).

> * Adds an {{elastic-defend}} advanced policy setting for Windows that lets you disable enrichment of DLL Search Order Hijacking detection events [#272068]({{kib-pull}}272068).

Reviewer 3: ```suggestion
* Adds an {{elastic-defend}} advanced policy setting for Windows that lets you disable enrichment of DLL Search Order Hijacking detection events [#272068]({{kib-pull}}272068).
* Adds cross-cluster search support for {{elastic-defend}} agents that use a remote Elasticsearch output. Endpoint management now reads host metadata, response actions, policy responses, field suggestions, and workflow insights from connected remote clusters [#271559]({{kib-pull}}271559), [#276749]({{kib-pull}}276749).
```

* Adds a new `canceled` response action status, shown in the response action history and console, with support across response action types including Microsoft Defender for Endpoint [#271357]({{kib-pull}}271357).

> * Raises the default file size limit for the `upload` response action to 500 MB and effectively removes the maximum, configurable via `xpack.securitySolution.maxUploadResponseActionFileBytes` [#275850]({{kib-pull}}275850).

Reviewer 4: We are dropping this to 200MiB, for sure, for the `9.5.0` release: https://github.com/elastic/kibana/pull/280227

Author: Per Slack conversation, keeping this as 500 MB since backport to 9.5 didn't merge before the final BC.

Author: Confirmed that the backport made it into the additional BC, so the change will be live in 9.5.0.

> * Raises the default file size limit for the `upload` response action to 500 MB and effectively removes the maximum, configurable via `xpack.securitySolution.maxUploadResponseActionFileBytes` [#275850]({{kib-pull}}275850).

Reviewer 4: ```suggestion
* Raises the default file size limit for the `upload` response action to 200 MiB and effectively removes the maximum, configurable via `xpack.securitySolution.maxUploadResponseActionFileBytes` [#275850]({{kib-pull}}275850).
```

* Fixes an issue where the prebuilt Privileged Users watchlist was only installed in the default space; it is now installed in custom spaces as well [#277645]({{kib-pull}}277645).
* Fixes an issue where entity maintainer tasks (risk score, watchlist, and relationship maintainers) kept running after the entity store was toggled off; they now stop when all engines stop and restart when engines are started again [#273890]({{kib-pull}}273890).
* Fixes the entities table on the **Entity Analytics** page to sort by risk score by default, showing the highest-risk entities first and placing entities without a risk score at the bottom [#272234]({{kib-pull}}272234).
* Fixes an issue on the **Entity Analytics** page where the search bar appeared blank after navigating to Discover and back, and where **Refresh** could reuse a stale query [#265926]({{kib-pull}}265926).
* Fixes an issue where the **Entity Analytics** left panel showed stale details when previewing a different entity [#265512]({{kib-pull}}265512).
* Fixes multiple rendering issues in the entity details flyout when opened in **Preview** mode from {{agent-builder}} [#267342]({{kib-pull}}267342).

> * Fixes a crash when adding an entity attachment for an entity with a single-value `watchlists` keyword field [#276201]({{kib-pull}}276201).

Reviewer 5: Can be removed, this feature will be customer-available only in 9.5, no need to mention this fix.

Author: Removed in [09157d8](https://github.com/elastic/docs-content/pull/7462/commits/09157d86b2878cd87a563be72957bb1cb3e79cf6).

* Fixes an issue where {{elastic-defend}} could lose Tamper Protection state on startup, by rehydrating the cache from disk when a policy is applied.
* Fixes {{elastic-defend}} recovery from network event subscription failures on macOS (26.5.2), where a failed subscription was previously sticky with no retry attempts.
* Fixes an {{elastic-defend}} crash that occurred when a Kafka broker rejected an oversized document.
* Fixes {{elastic-defend}} to properly honor {{es}} `queue.*` settings sent from {{kib}}, as well as queue presets.
* Fixes validation of the `bulk_max_size` output setting in {{elastic-defend}}.
* Fixes a race condition when assigning the username of mounted USB devices in {{elastic-defend}} on Windows.
* Improves the accuracy of serial number gathering in {{elastic-defend}} on Windows.
* Fixes a rare process tracking issue in {{elastic-defend}} on macOS and Linux.
* Fixes an issue where {{elastic-defend}} on Linux did not properly read tty events on older kernels (`4.18`) with eBPF backports, such as on RHEL 8.

> * Improves the reliability of fanotify event collection in {{elastic-defend}} on Ubuntu 26.04 and Linux 7.0.

Reviewer 6: Need something like this added:
```
{elastic-defend} no longer performs any file reads for executables matching a Trusted Application entry on Linux.
```

This slipped through because the release note was in the description of a [PR](https://github.com/elastic/endpoint-dev/pull/21016) stacked on another branch instead of main, and I neglected to carry it through to main [PR](https://github.com/elastic/endpoint-dev/pull/21006) after merging. I updated the main PR's release note to have the text above, as it applies to both.

Thank you @gogochan for noticing. 🙇

Author: Added in [09157d8](https://github.com/elastic/docs-content/pull/7462/commits/09157d86b2878cd87a563be72957bb1cb3e79cf6).

* Fixes {{elastic-defend}} on Windows to display the custom popup on cached denied requests in Device Control.
* Fixes {{elastic-defend}} handling of `windows.advanced.events.security.event_disabled` on Windows, so removing event IDs from the list correctly re-enables those Security events without requiring an additional policy reapply.
* Fixes a rare edge case where {{elastic-defend}} could lose Tamper Protection.
* Fixes an issue with rollback configuration in {{elastic-defend}} on Windows, where under certain runtime conditions the endpoint could get stuck in the `CONFIGURING` state for an extended period when updating a policy with rollback enabled.

> * Fixes a partial-write bug so {{elastic-defend}} correctly sends data to a {{ls}} output when the kernel TCP send buffer is full, resolving premature connection failures.

Reviewer 6: {{ls}} is logstash, right?

Author: Yep! There are links to the preview pages [here](https://github.com/elastic/docs-content/pull/7462#issuecomment-5023274724), if that helps.
